aloha-io-avro/src/main/scala/com/eharmony/aloha/factory/avro/factory_config.scala

```diff
@@ -0,0 +1,179 @@
+package com.eharmony.aloha.factory.avro
+
+import java.io.{File, InputStream}
+
+import com.eharmony.aloha.audit.impl.avro.{AvroScoreAuditor, Score}
+import com.eharmony.aloha.factory.ModelFactory
+import com.eharmony.aloha.factory.ex.AlohaFactoryException
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.semantics.compiled.CompiledSemantics
+import com.eharmony.aloha.semantics.compiled.compiler.TwitterEvalCompiler
+import com.eharmony.aloha.semantics.compiled.plugin.avro.CompiledSemanticsAvroPlugin
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.commons.io.IOUtils
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scala.util.{Failure, Success, Try}
+
+sealed abstract class FactoryConfig {
+  /**
+    * Attempt to create a factory.
+    * @return
+    */
+  def apply(): Try[ModelFactory[GenericRecord, Score]]
+
+  protected[this] def semantics(s: Schema,
+                              imports: Seq[String],
+                              classCacheDir: Option[File],
+                              dereferenceAsOptional: Boolean): CompiledSemantics[GenericRecord] = {
+    val p = CompiledSemanticsAvroPlugin[GenericRecord](s, dereferenceAsOptional)
+    CompiledSemantics(TwitterEvalCompiler(classCacheDir = classCacheDir), p, imports)
+  }
+
+  protected[this] def auditor[A](refInfo: RefInfo[A]): Try[AvroScoreAuditor[A]] =
+    AvroScoreAuditor(refInfo).map(Success.apply).getOrElse(
+      Failure(new AlohaFactoryException(
+        s"Couldn't create AvroScoreAuditor for ${RefInfoOps.toString(refInfo)}")))
+
+  protected[this] def refInfo(refInfoStr: String): Try[RefInfo[Any]] =
+    RefInfo.fromString(refInfoStr) match {
+      case Left(err) => Failure(new AlohaFactoryException(err))
+      case Right(success) => Success(success.asInstanceOf[RefInfo[Any]])
+    }
+
+  protected[this] def schemaInputStream(vfsUrl: String, useVfs2: Boolean): Try[InputStream] =
+    Try {
+      if (useVfs2)
+        org.apache.commons.vfs2.VFS.getManager.resolveFile(vfsUrl).getContent.getInputStream
+      else org.apache.commons.vfs.VFS.getManager.resolveFile(vfsUrl).getContent.getInputStream
+    }
+
+  protected[this] def schemaFromIs(is: InputStream): Try[Schema] =
+    Try { new Schema.Parser().parse(is) }
+
+  protected[this] def schema(vfsUrl: String, useVfs2: Boolean): Try[Schema] = {
+    val isTry = schemaInputStream(vfsUrl, useVfs2)
+    val inSchema = for {
+      is <- isTry
+      s <- schemaFromIs(is)
+    } yield s
+    isTry.foreach(IOUtils.closeQuietly)
+    inSchema
+  }
+}
+
+object FactoryConfig {
+
+  /**
+    * To enable the use of the magnet pattern, import `MagnetSyntax._`.
+    * See Spray's [[http://spray.io/blog/2012-12-13-the-magnet-pattern/ Magnet pattern]]
+    * article for more information.
+    */
+  object MagnetSyntax {
```

Comment:
Do we really want to support this?  Seems nice but might provide too many ways to do the same thing.